s4e-web/src/app/views/map-view/sentinel-search/sentinel-form/sentinel-form.component.ts

```diff
@@ -0,0 +1,136 @@
+import {Component, forwardRef, Input, OnDestroy, OnInit} from '@angular/core';
+import {
+  isSentinelFloatParam, isSentinelSelectParam,
+  SentinelParam,
+  SentinelSelectParam
+} from '../../state/sentinel-search/sentinel-search.metadata.model';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  FormControl,
+  FormGroup, NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  Validators
+} from '@angular/forms';
+import {disableEnableForm, stripNulls} from '../../../../utils/miscellaneous/miscellaneous';
+import {Subscription} from 'rxjs';
+
+
+@Component({
+  selector: 's4e-sentinel-form',
+  templateUrl: './sentinel-form.component.html',
+  styleUrls: ['./sentinel-form.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SentinelFormComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => SentinelFormComponent),
+      multi: true,
+    }
+  ]
+})
+export class SentinelFormComponent implements OnInit, OnDestroy, ControlValueAccessor, Validator {
+  form: FormGroup = new FormGroup({});
+  paramsDefPairs: ([SentinelParam, SentinelParam] | [SentinelParam])[];
+  private sub: Subscription|null = null;
+
+  @Input('paramsDef') set paramsDef(val: SentinelParam[]) {
+    if(this.sub) {
+      this.sub.unsubscribe();
+    }
+
+    const formControls: FormControl[] = [];
+    this.form = new FormGroup({});
+    this.paramsDefPairs = [];
+    val.forEach((formControlDef, i) => {
+      if(i % 2 == 1) {
+        this.paramsDefPairs.push([val[i-1], val[i]]);
+      }
+
+      let fc = new FormControl();
+      this.form.addControl(formControlDef.queryParam, fc);
+
+      if (isSentinelSelectParam(formControlDef)) {
+        fc.setValue(formControlDef.values[0]);
```

Comment:
What value is behind this 0 index? Could you move it to const and describe it better?

Reply:
this is just that - the control is a select box, so I set the value to the first option. But I'll add comment so that no one will have any doubts later